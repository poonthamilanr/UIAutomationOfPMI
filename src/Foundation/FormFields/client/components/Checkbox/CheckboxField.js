import React from 'react';
import { Field } from 'formik';
import { Checkbox } from '@pmi/dsm-react-bs4';
import { FieldError, FieldGroup, FieldLabel } from '../FieldGroup';
import { getFieldError, getElementId } from '../utils';
import './checkbox.scss';

export class CheckboxField extends React.PureComponent {
  renderField = ({ field, form }) => {
    const { label, linkText, onClickLink, fieldName, index, required, disabled, onChange, classNameGroup } = this.props;
    const { value, name } = field;
    const error = getFieldError(form, fieldName);
    const groupProps = { required, disabled, error, classNameGroup };

    const change = e => {
      field.onChange(e);
      if (onChange) {
        onChange({...form.values, [field.name]: e.target.checked });
      }
    };

    const id = getElementId('checkbox', fieldName, index);

    return (
      <FieldGroup {...groupProps}>
        <div className="checkbox-form-field">
          <Checkbox
            name={name}
            inputId={id}
            value={value}
            checked={value}
            onChange={change}
            ariaLabelledby={`${id}-label`}
          />
          {label &&
              <FieldLabel
                id={id}
                label={label}
                linkText={linkText}
                onClickLink={onClickLink}
                required={required}
                className="checkbox-label"
              />}
        </div>
        <FieldError error={error} fieldName={fieldName} index={index} />
      </FieldGroup>
    );
  };

  render() {
    return <Field name={this.props.fieldName} render={this.renderField} />;
  }
}
