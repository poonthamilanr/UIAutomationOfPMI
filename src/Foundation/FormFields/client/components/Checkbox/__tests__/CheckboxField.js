import React from 'react';
import { shallow } from 'enzyme';
import { CheckboxField } from '../CheckboxField';

describe('CheckboxField', () => {
  it('should render correctly', () => {
    const component = shallow(<CheckboxField />);

    expect(component).toMatchSnapshot();
  });
});