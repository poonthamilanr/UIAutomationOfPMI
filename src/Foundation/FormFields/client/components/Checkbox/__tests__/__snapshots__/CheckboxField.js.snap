// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxField should render correctly 1`] = `ShallowWrapper {}`;
